### PR TITLE
streaming arrow data support

### LINKED ISSUE #171: Support streaming query
DuckDB now supports [streaming queries](https://github.com/duckdb/duckdb/pull/6318) which do not materialize everything before returning the first rows

### PR DESCRIPTION
**Pull request by y-f-u**
This PR adds a new `execute_streaming` method in Statement and a `ArrowStream` struct to support streaming arrow data rather than fetching all in one call.

This has a caveat that require a SchemaRef to be passed into the `ArrowStream` struct as I couldn't find a way to get the schema without fetching entire batches.

Please point out if anyone knows

This fixes #171

### CODE REVIEW

**Comment by Mytherin** [comment]
The current CI failure looks to be related to the actual changes made in the PR. Could you have a look?

**Comment by peasee** [comment]
> The current CI failure looks to be related to the actual changes made in the PR. Could you have a look?

Should be fixed now! I've tested the workflow in our fork which passed 😃

**Comment by hozan23** [comment]
Hello, 

Any updates on this PR? @y-f-u @Mytherin

File: crates/duckdb/src/raw_statement.rs [hunk 1]
```diff
@@ -242,6 +277,22 @@ impl RawStatement {
         }
     }
 
+    pub fn execute_streaming(&mut self) -> Result<()> {
+        self.reset_result();
+        unsafe {
```
Comment (sfc-gh-vparakh) [inline crates/duckdb/src/raw_statement.rs:282]: best practice: limit the scope of the unsafe to the smallest section of code that needs it

**Pull request merged**